src/main/java/org/molgenis/vcf/utils/metadata/FieldMetadataServiceImpl.java

```diff
@@ -0,0 +1,186 @@
+        String description = jsonFieldMetadata.getDescription() != null ? jsonFieldMetadata.getDescription() : line.getDescription();
+        Map<String, NestedFieldMetadata> nestedFields = jsonFieldMetadata.getNestedFields() != null ? mapNestedFields(line, jsonFieldMetadatas) : null;
+        ValueDescription nullValue = jsonFieldMetadata.getNullValue() != null ? jsonFieldMetadata.getNullValue() : null;
+        return FieldMetadata.builder().label(line.getID()).description(description).type(type)
+                .numberType(numberType).numberCount(numberCount).categories(categories).separator(separator)
+                .label(label).nestedFields(nestedFields).nullValue(nullValue).required(required).build();
+    }
+
+    private FieldMetadata mapCustomFormatFieldMetadata(VCFCompoundHeaderLine line, JsonFieldMetadatas jsonFieldMetadatas) {
+        JsonFieldMetadata jsonFieldMetadata = jsonFieldMetadatas.getFormat().get(line.getID());
+        return mapCustomFieldMetadata(line, jsonFieldMetadatas, jsonFieldMetadata);
+    }
+
+    private Map<String, NestedFieldMetadata> mapNestedFields(VCFCompoundHeaderLine line, JsonFieldMetadatas jsonFieldMetadatas) {
+        Map<String, NestedFieldMetadata> nestedFields = new HashMap<>();
+        FieldType type = line instanceof VCFInfoHeaderLine ? INFO : FORMAT;
+        FieldIdentifier identifier = FieldIdentifier.builder().name(line.getID()).type(type).build();
+        if (!nestedAttributesMap.containsKey(identifier)) {
+            throw new MissingPrefixException(identifier);
+        }
+        NestedAttributes nestedAttributes = nestedAttributesMap.get(identifier);
+        String description = line.getDescription();
+        String escapedSeparator = Pattern.quote(nestedAttributes.getSeperator());
+        String[] infoIds = description.substring(nestedAttributes.getPrefix().length()).split(escapedSeparator, -1);
+        int index = 0;
+        for (String id : infoIds) {
+            NestedFieldMetadata.NestedFieldMetadataBuilder<?, ?> nestedFieldMetadataBuilder = NestedFieldMetadata
+                    .builder().label(id).index(index);
+            NestedJsonFieldMetadata jsonMeta = getNestedMetadata(line, jsonFieldMetadatas, id);
+            if (jsonMeta != null) {
+                nestedFieldMetadataBuilder.label(jsonMeta.getLabel()).description(jsonMeta
+                                .getDescription()).type(jsonMeta.getType())
+                        .numberType(mapNumberType(jsonMeta.getNumberType())).numberCount(jsonMeta.getNumberCount())
+                        .categories(jsonMeta.getCategories()).separator(jsonMeta.getSeparator())
+                        .nullValue(jsonMeta.getNullValue())
+                        .required(jsonMeta.getRequired() != null ? jsonMeta.getRequired() : false).build();
+            }else{
+                //No metadata available, assume non-required single string value
+                nestedFieldMetadataBuilder.type(ValueType.STRING)
```

Comment:
dangerous assumption: throw exception instead (unless this is the current behavior as well)?

Reply:
that would lead to exceptions for every field in the VEP annotation that does not have a metadata mapping...

Reply:
ok, lets keep like this then